File: TP1/travail-a-rendre/app/src/main/java/com/example/app1/MainActivity.java
package com.example.app1;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.RadioButton;
import android.widget.Spinner;
import android.widget.Toast;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

public class MainActivity extends AppCompatActivity {

    private EditText nomEdit;
    private EditText emailEdit;
    private EditText phoneEdit;
    private EditText adresseEdit;
    private Button send, reset;
    private Spinner villes;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_main);

        nomEdit = findViewById(R.id.nomEdit);
        emailEdit = findViewById(R.id.emailEdit);
        phoneEdit = findViewById(R.id.phoneEdit);
        adresseEdit = findViewById(R.id.adresseEdit);
        villes  =  findViewById(R.id.villes);
        send = findViewById(R.id.send);
        reset = findViewById(R.id.reset);


        reset.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                nomEdit.getText().clear();
                emailEdit.getText().clear();
                phoneEdit.getText().clear();
                adresseEdit.getText().clear();
                villes.setSelection(0);
            }
        });
        send.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {

                Intent intent = new Intent(MainActivity.this, SecondActivity.class);
                intent.putExtra("nom", nomEdit.getText().toString());
                intent.putExtra("email", emailEdit.getText().toString());
                intent.putExtra("phone", phoneEdit.getText().toString());
                intent.putExtra("adresse", adresseEdit.getText().toString());
                intent.putExtra("ville", villes.getSelectedItem().toString());

                startActivity(intent);
            }

        }

        );
    }
}

File: Projet_Controle_NO_ROOM/app/src/main/res/layout/project/MainActivity.kt
package ma.ensa.project

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.viewpager2.widget.ViewPager2
import com.google.android.material.tabs.TabLayout
import com.google.android.material.tabs.TabLayoutMediator
import ma.ensa.project.R
import ma.ensa.project.view.adapters.ViewPagerAdapter
class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.viewpager_layout)

        val tabLayout = findViewById<TabLayout>(R.id.tabLayout)
        val viewPager = findViewById<ViewPager2>(R.id.viewPager)

        val adapter = ViewPagerAdapter(supportFragmentManager, lifecycle)
        viewPager.adapter = adapter

        TabLayoutMediator(tabLayout, viewPager) { tab, position ->
            tab.text = when (position) {
                0 -> "Login"
                1 -> "Register"
                else -> "Login"
            }
        }.attach();
    }
}